rllib/algorithms/algorithm_config.py

```diff
@@ -925,10 +932,12 @@ def evaluation(
             self.evaluation_num_workers = evaluation_num_workers
         if custom_evaluation_function is not None:
             self.custom_evaluation_function = custom_evaluation_function
-        if always_attach_evaluation_results:
+        if always_attach_evaluation_results is not None:
```

Comment:
I noticed that these are buggy. i.e. previously if always_attach_evaluation_results was set to False and by default it was true this call would not have overriden it.

Reply:
You have to explicitly check if these variables are not None otherwise False would also not get assigned